### PR TITLE
[FLINK-31412] HiveCatalog rollbacks schema in dfs if it alters table failed

### PR DESCRIPTION
**Pull request by FangYongs**
This PR amis to delete updated schema in dfs when `HiveCatalog` updates table failed in `metastore`

### CODE REVIEW

**Comment by FangYongs** [comment]
E2e test fails du to known issue

File: flink-table-store-hive/flink-table-store-hive-catalog/src/main/java/org/apache/flink/table/store/hive/HiveCatalog.java [hunk 1]
```diff
@@ -297,13 +297,20 @@ public void alterTable(
         }
 
         try {
+            final SchemaManager schemaManager = schemaManager(identifier);
             // first commit changes to underlying files
             TableSchema schema = schemaManager(identifier).commitChanges(changes);
```
Comment (JingsongLi) [inline flink-table-store-hive/flink-table-store-hive-catalog/src/main/java/org/apache/flink/table/store/hive/HiveCatalog.java:302]: use `schemaManager` directly.

File: flink-table-store-hive/flink-table-store-hive-catalog/src/main/java/org/apache/flink/table/store/hive/HiveCatalog.java [hunk 2]
```diff
@@ -297,13 +297,20 @@ public void alterTable(
...
-            updateHmsTable(table, identifier, schema);
-            client.alter_table(identifier.getDatabaseName(), identifier.getObjectName(), table);
+            try {
+                // sync to hive hms
+                Table table =
+                        client.getTable(identifier.getDatabaseName(), identifier.getObjectName());
+                updateHmsTable(table, identifier, schema);
+                client.alter_table(identifier.getDatabaseName(), identifier.getObjectName(), table);
+            } catch (TException te) {
+                fileIO.deleteQuietly(schemaManager.toSchemaPath(schema.id()));
```
Comment (JingsongLi) [inline flink-table-store-hive/flink-table-store-hive-catalog/src/main/java/org/apache/flink/table/store/hive/HiveCatalog.java:311]: introduce a function: `SchemaManager.deleteSchema(long schemaId)`?

Comment (FangYongs) [reply flink-table-store-hive/flink-table-store-hive-catalog/src/main/java/org/apache/flink/table/store/hive/HiveCatalog.java]: Done

File: flink-table-store-hive/flink-table-store-hive-catalog/src/main/java/org/apache/flink/table/store/hive/HiveCatalog.java (hunk 1 above)
Comment (FangYongs) [reply flink-table-store-hive/flink-table-store-hive-catalog/src/main/java/org/apache/flink/table/store/hive/HiveCatalog.java]: Done

**Review by JingsongLi — APPROVED** [review]
+1

**Pull request merged**